bundles/io.github.linkedfactory.service/src/main/java/io/github/linkedfactory/service/config/KvinPartitionedFactory.java

```diff
@@ -0,0 +1,48 @@
+package io.github.linkedfactory.service.config;
+
+import io.github.linkedfactory.core.kvin.Kvin;
+import io.github.linkedfactory.core.kvin.partitioned.KvinPartitioned;
+import net.enilink.composition.annotations.Iri;
+import org.eclipse.core.runtime.Platform;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+
+@Iri("plugin://io.github.linkedfactory.service/data/KvinPartitioned")
+public abstract class KvinPartitionedFactory implements IKvinFactory{
+    private static final Logger log = LoggerFactory.getLogger(KvinPartitionedFactory.class);
+
+    @Override
+    public Kvin create() {
+        try {
+            String path = getArchivePath();
+            Integer cycle = getCycle();
+            if (path == null) {
+                path = "partition";
+            }
+            if(cycle == null) {
+                cycle = 0;
+            }
+                File archivePath;
+            if (new File(path).isAbsolute()) {
+                archivePath = new File(path);
+            } else {
+                archivePath = new File(new File(Platform.getInstanceLocation().getURL().toURI()), path);
+            }
+            log.info("Using path: {} for archiving", archivePath);
+
+            return new KvinPartitioned(archivePath, cycle);
+        } catch (IOException | URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Iri("plugin://io.github.linkedfactory.service/data/archivePath")
+    public abstract String getArchivePath();
+
+    @Iri("plugin://io.github.linkedfactory.service/data/cycle")
+    public abstract Integer getCycle();
```

Comment:
If this is mapped to `int` then `0` should be returned automatically if value is `null`.